usr/src/uts/common/io/overlay/overlay.c

```diff
@@ -1107,7 +1107,8 @@ overlay_m_tx(void *arg, mblk_t *mp_chain)
 	mutex_enter(&odd->odd_lock);
 	overlay_io_done(odd, OVERLAY_F_IN_TX);
 	mutex_exit(&odd->odd_lock);
-	return (mp_chain);
+	freemsgchain(mp_chain);
```

Comment:
It looks like everywhere that calls `overlay_m_tx()` calls `freemsgchain()`.  Should those other calls be removed or does it not matter since it's checking for != NULL anyways?

Reply:
overlay_m_tx() is a mc_tx(9e) function, it has clear semantics about NULL/non-NULL returns.  That man page is good background reading about why there are two different proposed solutions.

And freemsgchain() will easily handle NULL for input.  I'd keep them in place to minimize diffs, and it'll allow for the "flowcontrol" solution to DTRT more easily.